examples/server/server.cpp

```diff
@@ -3691,37 +3692,18 @@ int main(int argc, char ** argv) {
             return;
         }
 
+        if (type == SERVER_TASK_TYPE_SLOT_SAVE) {
+            GGML_ASSERT(dynamic_cast<server_task_result_slot_save_load*>(result.get()) != nullptr);
```

Comment:
missing check for `SERVER_TASK_TYPE_SLOT_RESTORE`

Reply:
you I'm not sure if we can use `template` here to avoid having a long list of `if..else`

Reply:
I'll take a look at this :+1:

Reply:
I've added a suggestion for this in https://github.com/ggerganov/llama.cpp/pull/11535/commits/6cc2956f3fa6f387c1b4c2b99dc42720623513f0.  

This contain an addition of a field, `server_task_type`,  and a virtual function in `server_task_result`. The idea being that the dynamic cast like this can be avoided:
```c++
GGML_ASSERT(dynamic_cast<server_task_result_slot_save_load*>(result.get()) != nullptr);
```
In favor of checks like this:
```c++
        GGML_ASSERT(result.get() != nullptr);                                       
        GGML_ASSERT(result.get()->get_server_task_type() == type);
```
If this looks like alright then I can update the other places in `server.cpp` which also have similar dynamic_casts.